core/drag/drag-manager.js

```diff
@@ -0,0 +1,1314 @@
+                            deltaPointerBottom = (
+                                this._draggedImageBoundingRect.bottom - 
+                                draggingOperationContext.startPositionY
+                            )
+                        ) > rect.bottom) {
+                            translateY = (
+                                rect.bottom - 
+                                draggingOperationContext.startPositionY - 
+                                deltaPointerBottom
+                            );
+                        }
+                    }
+
+                    var translate = "translate3d(";
+                    translate += translateX;
+                    translate += "px,";
+                    translate += translateY;
+                    translate += "px,0)";
+
+                    draggedImage.style[DragManager.cssTransform] = translate;
+
+                    this._scrollIfNeeded(
+                        draggingOperationContext.positionX, 
+                        draggingOperationContext.positionY
+                    );
+                }
+
+                this._rootComponent.element.style.cursor = this._cursorStyle;
+
+            } else if (this._willTerminateDraggingOperation) {
+                this._rootComponent.element.style.cursor = "default";
+
+                if (draggingOperationContext.draggedImage) {
+                    document.body.removeChild(draggingOperationContext.draggedImage);
+                }
+
+                if (draggingOperationContext.currentDropTarget) {
+                    draggingOperationContext.hasBeenDrop = true;
+                    draggingOperationContext.droppable = draggingOperationContext.currentDropTarget;
+                }
+
+                this._dispatchDrop(
+                    draggingOperationContext
+                );
+                
+                this._resetTranslateContext();
+                draggingOperationContext.currentDropTarget = null;
+                this._dispatchDragEnd(draggingOperationContext);
+
+                if (
+                    draggingOperationContext.draggable &&
+                    draggingOperationContext.dragEffect === "move"
+                ) {
+                    this._shouldRemovePlaceholder = true;
+                    this._rootComponent.needsDraw = true;
+                    // Wait for the next draw cycle to remove the placeholder,
+                    // or in order to be synchronised with the draw cyle when 
+                    // the draggable component will become visible again.
+                    // Plus it allows the receiver to perform any necessary clean-up. 
+                    return void 0;
+                } else {
+                    this._draggingOperationContext = null;
+                }
+            }
+
+            this._removeDraggablePlaceholderIfNeeded();
+        }
+    },
+
+    _setUpDraggedImageIfNeeded: {
```

Comment:
lets call this _drawDraggedImageIfNeeded instead, that way it's clear this is is a piece of the draw